sys/kern/pool.c

```diff
@@ -196,32 +203,31 @@ void *pool_alloc(pool_t *pool, unsigned flags) {
  * (maybe leave one) */
 void pool_free(pool_t *pool, void *ptr) {
   debug("pool_free: pool = %p, ptr = %p", pool, ptr);
-
   assert(pool->pp_state == ALIVE);
 
   /* Mark the item as invalid */
   kasan_mark(ptr, 0, pool->pp_itemsize, KASAN_CODE_POOL_USE_AFTER_FREE);
 
-  WITH_MTX_LOCK (&pool->pp_mtx) {
```

Comment:
Is this change really necessary?

Reply:
In quarantine PR, this function will be renamed to `_pool_free` and called from quarantine code (if KASAN is enabled) or from `pool_free` (if KASAN is disabled) **with mutex already acquired**, so this `WITH_MTX_LOCK (&pool->pp_mtx)` will be deleted anyway. I wanted to move this code change from that PR to this one, as it does not introduce any new features.

Reply:
Ok.